### PR TITLE
Fix: vue2 events use bracket notation

### PR DESCRIPTION
**Pull request by fazulk**
Issue: #20753

## What I did

~~Fix casing for vue2 events to be camelcased by default when using the default storybook render fn.~~
Allow kebab case event names to work correctly in vue. 

## How to test

1. load vue2/vite sandbox. 
2. Add `this.$emit('also-clicked')` to line 51 of Button.vue
3. Remove the render function, line 8-12 from Button.stories.js
4. Add arg types `'also-clicked': { action: 'clicked' }` to argTypes
5. Error should be gone and should show event inside storybook actions.

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [x] Make sure to add/update documentation regarding your changes
- [x] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

File: code/renderers/vue/src/render.ts [hunk 1]
```diff
@@ -81,7 +83,7 @@ export const render: ArgsStoryFn<VueRenderer> = (args, context) => {
     .map((argType) => argType.name);
 
   if (eventProps.length) {
-    eventsBinding = eventProps.map((name) => `@${name}="$props.${name}"`).join(' ');
+    eventsBinding = eventProps.map((name) => `@${name}="$props.${camelCase(name)}"`).join(' ');
```
Comment (ndelangen) [inline code/renderers/vue/src/render.ts:86]:
```suggestion
    eventsBinding = eventProps.map((name) => `@${name}="$props['${name}"']`).join(' ');
```

File: code/renderers/vue/src/render.ts [hunk 2]
```diff
@@ -49,6 +49,8 @@ const getRoot = (canvasElement: VueRenderer['canvasElement']): Instance => {
   return instance;
 };
 
+const camelCase = (str: string) => str.replace(/-([a-z])/g, (g) => g[1].toUpperCase());
```
Comment (ndelangen) [inline code/renderers/vue/src/render.ts:52]:
```suggestion
```

**Comment by ndelangen** [comment]
@fazulk thank you for contributing, we discussed this with the team, and we think a better solution would be to use bracket-notation to reference the property, instead of trying to convert it to something it might or might not be. It's less magic.

I left 2 comments with suggestion, could you have a look at applying them, and test if this indeed is a viable alternative? 🙏

**Review by ndelangen — APPROVED** [review]

File: code/renderers/vue/src/render.ts [hunk 3]
```diff
@@ -81,7 +82,7 @@ export const render: ArgsStoryFn<VueRenderer> = (args, context) => {
     .map((argType) => argType.name);
 
   if (eventProps.length) {
-    eventsBinding = eventProps.map((name) => `@${name}="$props.${name}"`).join(' ');
+    eventsBinding = eventProps.map((name) => `@${name}="$props['${name}"']`).join(' ');
```
Comment (phunguyenmurcul) [inline code/renderers/vue/src/render.ts:85]: @ndelangen As far as I know Vue2 will convert kebab-case prop to camelCase props. so if you define `also-clicked` in `argTypes` or story `args`, you will see only `alsoClicked` in `props`

Screenshot at Feb 01 22-03-39
Screenshot at Feb 01 22-03-15

**Comment by fazulk** [comment]
> @fazulk thank you for contributing, we discussed this with the team, and we think a better solution would be to use bracket-notation to reference the property, instead of trying to convert it to something it might or might not be. It's less magic.
> 
> I left 2 comments with a suggestion, could you have a look at applying them, and test if this indeed is a viable alternative? 🙏

@ndelangen , unfortunately that doesn't work. Because vue2 converts props from kebab-case to camelCase. the prop 'also-clicked' will be undefined, as @phunguyenmurcul pointed out. I tried the method you mentioned, by using `$props['some-event']` and it didn't work. Let me know if I should re-commit the changes made, or if you'd like to explore another solution, But because prop conversion to camel case is baked into vue -- i'm not sure if there is another route.

**Comment by ndelangen** [comment]
OK, then yeah, we should change back to the code you committed before, sorry to make this more difficult then it needed to be.

**Comment by fazulk** [comment]
@ndelangen Made an adjustment here which doesn't require mutating the key name of the event. 

Using the mutation to camelCase worked, but there was console errors for any events that weren't declared in `argTypes` or caught by the `argTypesRegex`. If you declared those, the errors went away. But you still needed to change them to camel case.

The work around is to check to see if the event is included in `$props` and not undefined. Then execute that function on the event. This works 100% of the time with `camelCasedEvents`

If that fails, then execute the function from initialArgs, which retains the original casing of the event. 

This allows the user to emit `this.$emit('something-happened')` and if our `argTypesRegex` isn't setup to catch it. The user can declare it:

```
argTypes: {
   'something-happened': { action: 'it really did' }
}
```

**Comment by ndelangen** [comment]
@storybookjs/vue I'm inclined to merge this PR, can someone leave a review 🙏 Thanks!

**Comment by fazulk** [comment]
Any update on when this might get looked at? This bug is the only thing thats preventing me from updating to storybook 7 ..

**Comment by chakAs3** [comment]
> Any update on when this might get looked at? This bug is the only thing thats preventing me from updating to storybook 7 ..

hi @fazulk  i will be reviewing your PR. we will sort out this today

**Review by chakAs3 — CHANGES_REQUESTED** [review]

File: code/renderers/vue/src/render.ts [hunk 4]
```diff
@@ -81,15 +81,27 @@ export const render: ArgsStoryFn<VueRenderer> = (args, context) => {
     .map((argType) => argType.name);
 
   if (eventProps.length) {
-    eventsBinding = eventProps.map((name) => `@${name}="$props.${name}"`).join(' ');
```
[the remark this answers is not in the document]
Comment (fazulk) [reply code/renderers/vue/src/render.ts:84]: No problem, this was my original commit actually :)

**Review by chakAs3 — APPROVED** [review]

**Pull request merged**